docs/friends.md

```diff
@@ -0,0 +1,100 @@
+
+When a friend request is sent or the user is added an in-app notification will be sent. See the [in-app notification](in-app-notifications.md#receive-notifications) section for more info.
+
+```csharp fct_label="Unity"
+byte[] userId = ...; // some user ID
+var message = NFriendAddMessage.Default(userId);
+client.Send(message, (bool done) => {
+  Debug.Log("Friend added or request sent.");
+}, (INError err) => {
+  Debug.LogErrorFormat("Error: code '{0}' with '{1}'.", err.Code, err.Message);
+});
+```
+
+When both users have added eachother as friends they are able to initiate realtime chat in a 1-on-1 channel. See the [realtime chat](realtime-chat.md) section for more info.
+
+## List friends
+
+You can list all of a user's friends, blocked users, friend requests received (invited), and invites they've sent. These statuses are returned together as part of the friend list which makes it easy to display in a UI.
+
+```csharp fct_label="Unity"
+var message = NFriendsListMessage.Default();
+client.Send(message, (INResultSet<INFriend> list) => {
+  foreach (var f in list.Results) {
+    var id = Encoding.UTF8.GetString(f.Id); // convert byte[].
+    // f.State is one of: Friend, Invite, Invited, Blocked.
+    Debug.LogFormat("User {0} has state {1}.", id, f.State);
+  }
+}, (INError err) => {
+  Debug.LogErrorFormat("Error: code '{0}' with '{1}'.", err.Code, err.Message);
+});
+```
+
+## Remove friends
+
+A user can remove a friend, reject a received invite, cancel a friend request sent, or unblock a user. Similar to how Friend Add works we reuse Friend Remove to cancel or undo whatever friend state is current with another user.
+
+!!! note
+    If a user is unblocked they are removed from the friend list entirely. To re-add them each user must add the other again.
+
+```csharp fct_label="Unity"
+byte[] userId = ...; // some user ID
+var message = NFriendRemoveMessage.Default(userId);
+client.Send(message, (bool done) => {
+  var idString = Encoding.UTF8.GetString(userId); // convert byte[].
+  Debug.Log("User {0} has been removed.", idString);
+}, (INError err) => {
+  Debug.LogErrorFormat("Error: code '{0}' with '{1}'.", err.Code, err.Message);
+});
+```
+
+## Block a friend
+
+You can stop a user from 1-on-1 chat or other social features with a user if you block them. The user who wants to block should send the message. They can be unblocked later with a [Friend Remove](#friend-remove) message.
+
+A user who has been blocked will not know which users have blocked them. That user can continue to add friends and interact with other users.
+
+```csharp fct_label="Unity"
+byte[] userId = ...; // some user ID
+var message = NFriendBlockMessage.Default(userId);
+client.Send(message, (bool done) => {
+  var idString = Encoding.UTF8.GetString(id); // convert byte[].
+  Debug.Log("User {0} has been blocked.", idString);
+}, (INError err) => {
+  Debug.LogErrorFormat("Error: code '{0}' with '{1}'.", err.Code, err.Message);
+});
+```
+
+### Ban a user
```

Comment:
@AlimJaffer It wouldn't make sense to incorporate advice about shadow bans with friend lists. It'd be something we can cover in the future with leaderboards and other parts of the system.